tests/operators/test_misc.py

```diff
@@ -195,6 +195,20 @@ def test_difference_incorrect_data_longitude_shape(cube):
         misc.difference(cubes)
 
 
+def test_difference_incorrect_data_shape_regrid(cube):
+    """Test when data shape differs, but gets regridded.
+
+    For a cube if it meets the requirements of the variable exceptions.
+    """
+    rearranged_cube = cube.copy()
+    rearranged_cube = rearranged_cube[:, :, 1:]
+    rearranged_cube.attributes["STASH"] = "m01s03i225"
+    other_cube = cube.copy()
+    del other_cube.attributes["cset_comparison_base"]
+    cubes = iris.cube.CubeList([rearranged_cube, other_cube])
+    assert type(misc.difference(cubes)) is iris.cube.Cube
```

Comment:
```suggestion
    difference = misc.difference(cubes)
    assert isinstance(difference, iris.cube.Cube)
    assert difference.shape == cube.shape
    assert difference.shape != rearrange_cube.shape
```

The reason we are regridding is so the shapes of the cubes match, so that is a reasonable thing to check.